letsencrypt/client/display/ops.py

```
@@ -0,0 +1,139 @@
+"""Contains UI methods for LE user operations."""
+import os
+import sys
+
+import zope.component
+
+from letsencrypt.client import interfaces
+from letsencrypt.client.display import util as display_util
+
+# Define a helper function to avoid verbose code
+util = zope.component.getUtility  # pylint: disable=invalid-name
+
+
+def choose_authenticator(auths, errs):
+    """Allow the user to choose their authenticator.
+
+    :param list auths: Where each of type
+        :class:`letsencrypt.client.interfaces.IAuthenticator` object
+    :param dict errs: Mapping IAuthenticator objects to error messages
+
+    :returns: Authenticator selected
+    :rtype: :class:`letsencrypt.client.interfaces.IAuthenticator`
+
+    """
+    descs = [auth.description if auth not in errs
+             else "%s (Misconfigured)" % auth.description
+             for auth in auths]
+
+    while True:
+        code, index = util(interfaces.IDisplay).menu(
+            "How would you like to authenticate with the Let's Encrypt CA?",
+            descs, help_label="More Info")
+
+        if code == display_util.OK:
+            return auths[index]
+        elif code == display_util.HELP:
+            if auths[index] in errs:
+                msg = "Reported Error: %s" % errs[auths[index]]
+            else:
+                msg = auths[index].more_info()
+            util(interfaces.IDisplay).notification(
+                msg, height=display_util.HEIGHT)
+        else:
+            sys.exit(0)
```

could we please raise a meaningful exception instead?